### ISSUE TITLE
Update Condvar docs to use match ergonomics

### ISSUE THREAD

**Issue by KevinWMatthews** [rust-lang/rust#62857]
In `std::sync::Condvar`'s [docs](https://doc.rust-lang.org/std/sync/struct.Condvar.html), it looks like [line 39](https://github.com/rust-lang/rust/blob/0c7cf8c633b388d0d8cc4991d41a281d6ddd47bd/src/libstd/sync/condvar.rs#L39):

```rust
let &(ref lock, ref cvar) = &*pair2;
```

 and [line 51](https://github.com/rust-lang/rust/blob/0c7cf8c633b388d0d8cc4991d41a281d6ddd47bd/src/libstd/sync/condvar.rs#L51)

```rust
let &(ref lock, ref cvar) = &*pair;
```

 were written [before](https://github.com/rust-lang/rust/commit/d3b8f56ae75a3dad07da0d685ffb16b51c486d68) the match ergonomics RFC [was merged](https://github.com/rust-lang/rfcs/pull/2005).

Is it reasonable/preferable to change these lines to leverage match ergonomics? They would become:

```rust
let (lock, cvar) = &*pair2;
```

and

```rust
let (lock, cvar) = &*pair
```

respectively.

**Comment by jonas-schievink** [comment]
Sounds like a good idea to me. This should probably also be done for *all* the examples on that type (there's quite a lot using this pattern). Want to send a PR?

**Comment by KevinWMatthews** [comment]
@jonas-schievink Sure! It will give me an excuse to read all of the `Condvar` documentation in detail.

**Comment by Amanieu** [comment]
Fixed by #62942
